edi_sale_oca/tests/common.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Camptocamp SA
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields
+
+
+class OrderMixin(object):
+    @classmethod
+    def _create_sale_order(cls, **kw):
+        """Create a sale order
+
+        :return: sale order
+        """
+        model = cls.env["sale.order"]
+        vals = dict(commitment_date=fields.Date.today())
+        vals.update(kw)
+        so_vals = model.play_onchanges(vals, [])
+        if "order_line" in so_vals:
+            so_vals["order_line"] = [(0, 0, x) for x in vals["order_line"]]
+        return model.create(so_vals)
+
+    @classmethod
+    def _setup_order(cls, **kw):
+        cls.product_a = cls.env.ref("product.product_product_4")
```

Comment:
Please, create products instead of using odoo demo ones, we want to have control about their configuration.
It also help with understanding during reviews, no one can guess what's their `sale_delay` without having to check odoo demo data.

Reply:
We can do this in a 2nd iteration